styx-common/src/main/java/com/spotify/styx/storage/InMemStorage.java

```diff
@@ -325,6 +328,7 @@ public void deleteActiveState(WorkflowInstance workflowInstance) {
 
   public Optional<Long> getCounterFromActiveStates(WorkflowInstance workflowInstance)
       throws IOException {
-    return Optional.ofNullable(activeStatesMap.get(workflowInstance));
+    return Optional.ofNullable(activeStatesMap.get(workflowInstance))
+        .map(PersistentWorkflowInstanceState::counter);
```

Comment:
I think `getLatestStoredCounter()` from AggregateStorage (and InMemStorage) should read from ActiveStates instead of from the events, especially now that we are migrating away from events as source of truth.

Reply:
`getLatestStoredCounter` is used when re-initializing a WFI that has been deactivated and removed from datastore. Thus we need to count the events in bigtable.

I'd be all for keeping WFI information/state in datastore even after deactivation, but not sure that should go into this PR.

Reply:
Let's not include this work in this PR, but I agree we should eventually remove such operational requirement from BigTable.